src/serialization.rs

```diff
@@ -4,15 +4,17 @@ use serde::ser::Serialize;
 use serde_json::{from_str, to_string, Value};
 use serde_json::map::Map;
 
-use errors::{Result};
+use errors::Result;
 use header::Header;
 
 
 /// The return type of a successful call to decode
 #[derive(Debug)]
-pub struct TokenData<T: Deserialize> {
+pub struct TokenData<T>
+    where for<'a> T: Deserialize<'a>
```

Comment:
This bound should not be here. Structs in Rust should never have trait bounds. Trait bounds are for behavior.

```rust
impl<T> Data<T> where T: Behavior { /* behavior, okay */ }

fn behave<T>() where T: Behavior { /* also behavior, also okay */ }

struct Data<T> where T: Behavior { /* don't do this */ }
```

The only exception is things like [`Cow`](https://doc.rust-lang.org/std/borrow/enum.Cow.html) that use associated types to define data.